src/main/java/htsjdk/samtools/liftover/LiftOver.java

```diff
@@ -115,12 +126,21 @@ public Interval liftOver(final Interval interval, final double liftOverMinMatch)
         double minMatchSize = liftOverMinMatch * interval.length();
 
         // Find the appropriate Chain, and the part of the chain corresponding to the interval to be lifted over.
+        int maxIntersectionSize = 0;
         for (final Chain chain : chains.getOverlaps(interval)) {
             final TargetIntersection candidateIntersection = targetIntersection(chain, interval);
             if (candidateIntersection != null && candidateIntersection.intersectionLength >= minMatchSize) {
                 if (chainHit != null) {
-                    // In basic liftOver, multiple hits are not allowed.
-                    return null;
+                    if (useMaxOverlap) {
```

Comment:
I don't think this works. in particular, while you are saving the `maxIntersectionSize`, you are not keeping the actual `candidateIntersection` around, which means that you will always end up using the last `candidateIntersection` that passes the `minMatchSize`. please add tests in which the maximal intersection is the middle of three valid (i.e. long enough) intersections.